contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -9,6 +9,28 @@ Feature gates are intended to cover the development life cycle of a feature - th
 Features generally progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end up deciding that a feature is not going to be supported and we end up marking them as `Deprecated`.
 
 The majority of features will go through all three stages, but occasionally there are features which may skip stages.
+While some exceptions may happen, approvers should use the following guidance:
+- features that involve API changes must progress through all `Alpha`, `Beta`, `GA` stages
```

Comment:
In general when we use this term I think it refers to PRs that would trigger API review. So, this is about the K8s APIs, not the broader user surface. So, basically, anything that makes you read https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md

Reply:
> do we consider cli flags and configs for kubelet to be a API change

Yes. https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md#what-parts-of-a-pr-are-api-changes is a crisp list of "things that are APIs"

APIs are the way things outside the project integrate with our binaries / servers, and changes to that surface area can disrupt or break integrators. It's important that we be very confident in things that increase that surface area, so having at least a single release where we are still able to adjust that surface area is important.

> So any new flag/api change to KubeletConfig would require alpha -> beta -> GA?

Kubelet config overall is still beta (:-/) so a field inside it can't really be more mature than beta, but generally, yes, new things start as alpha for a release.

Reply:
I think the other important point is the removal of flags, removing a flag in kubelet per example will make that the new version will fail to start with "unknown flag" (I can not remember the exactly)  so removal of flags need to consider this problem

Reply:
> So, this is about the K8s APIs, not the broader user surface. So, basically, anything that makes you read https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md

we should starting thinking about moving these or some of this conventions to  the CRI API too

Reply:
Added a suggestion with Jordan's link